integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/viewpopenqueueservice/test/ViewPopEnqueueTest.java

```diff
@@ -0,0 +1,120 @@
+
+    private MessageStoreAdminClient messageStoreAdminClient;
+    MessageProcessorAdminService messageProcessorAdminService;
+    private JMSQueueMessageProducer jmsQueueMessageProducer;
+
+    private final String STORE_NAME = "VPE_Store";
+    private final String PROCESSOR_NAME = "VPE_Processor";
+    private final String PROXY_NAME = "VPE_Proxy";
+
+    /**
+     *  Initializing environment variables
+     */
+
+    @BeforeClass(alwaysRun = true, description = "Test Message store View, Pop and Enqueue")
+    protected void setup() throws Exception {
+        super.init();
+
+        loadESBConfigurationFromClasspath("artifacts/ESB/messageProcessorConfig/viewPopEnqueueTestConfig.xml");
+
+        // Initialization
+        messageStoreAdminClient = new MessageStoreAdminClient(context.getContextUrls().getBackEndUrl(), sessionCookie);
+
+        // Verifying Existence
+        verifyMessageStoreExistence(STORE_NAME);
+        verifyMessageProcessorExistence(PROCESSOR_NAME);
+        isProxySuccesfullyDeployed(PROXY_NAME);
+        //verifySequenceExistence(PROXY_NAME);
+
+    }
+
+    /**
+     *  1. Invoke the proxy with one request while the backend is unavailable
+     *  2. Verify the existence of messages in message store + automatic de-activation of processor due to backend failure
+     *  3. Call getMessage function and verify that the queue is sending the right message
+     *
+     *  Run View and Pop functions
+     *  Assert
+     */
+
+    @Test(groups = {"wso2.esb"}, description = "Test View and Pop and service in MessageStore")
+    public void testViewInMessageStore() throws Exception {
+
+
+        //Invoke proxy to send message to store
+        //axis2Client.sendSimpleQuoteRequest(getProxyServiceURLHttp(PROXY_NAME), null, "IBM");
+
+
+        String inputPayload = "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\">\n"
+                + "<soapenv:Header/>\n"
+                + "<soapenv:Body>\n"
+                + "<m0:getQuote xmlns:m0=\"http://services.samples\">\n"
+                + " <m0:request>IBM\n"
+                + " </m0:request>\n"
+                + "   <m0:request>WSO2\n"
+                + " </m0:request>\n"
+                + "</m0:getQuote>\n"
+                + "</soapenv:Body>\n"
+                + "</soapenv:Envelope>";
+
+        Map<String, String> requestHeader = new HashMap<>();
+        requestHeader.put("Content-type", "text/xml");
+        requestHeader.put("SOAPAction", "urn:mediate");
+        requestHeader.put("Accept", "application/json");
+
+        HttpRequestUtil.doPost(new URL(getProxyServiceURLHttp(PROXY_NAME)), inputPayload, requestHeader);
+
+        //Check if the messages have been sent to the Store
+        MessageInfo[] messages = messageStoreAdminClient.getPaginatedMessages(STORE_NAME, 0);
+
+        Assert.assertEquals(messages.length, 0, "Pending message count " + messages.length + " from list for Message store : " +
```

Comment:
Are we assuming that the message will be picked up by the processor just after adding it to the store? I think it might not be.